File: tech-dhtml/connect4/views_olds/TurnView.js
import { PlayerVisitor } from '../models/Player.js';
import { HumanPlayerView, RandomPlayerView } from './PlayerView.js';
//import { InIntervalDialog } from '../utils/views/Dialog.js';
import { Message } from './Message.js';
import { ColorView } from './BoardView.js';


class TurnView extends PlayerVisitor {
    #turn;
    #activePlayerView;
    #turnViewDiv;

    constructor(turn) {
        super();
        this.#turn = turn;
        this.#turnViewDiv=document.getElementById("turnViewDiv");
    }
    setNumberOfHumanPlayers() {
        /*  let inIntervalDialog = new InIntervalDialog(0, this.#turn.getNumberPlayers());
          inIntervalDialog.read(Message.NUM_PLAYERS.toString());*/

        this.#turn.reset(0); /*inIntervalDialog.getAnswer()*/
    }

    async play() {
        this.#turn.getActivePlayer().accept(this);
        this.write();
        this.#turn.play(await this.#activePlayerView.getColumn());
    }

    write(){
         new Message(Message.TURN + new ColorView(this.#activePlayerView.getPlayer().getColor()).toString()).writeln(this.#turnViewDiv);
    }

    writeResult() {
        if ((this.#turn.getBoard()).isWinner()) {
            this.writeWinner();
        } else {
            Message.PLAYERS_TIED.writeln(this.#turnViewDiv);
        }
    }

    writeWinner() {
        let message = Message.PLAYER_WIN.toString();
        message = message.replace(`#color`, new ColorView(this.#turn.getActivePlayer().getColor()).toString());
        new Message(message).writeln(this.#turnViewDiv);
    }

    

    visitHumanPlayer(humanPlayer) {
        this.#activePlayerView = new HumanPlayerView(humanPlayer);
    }
    visitRandomPlayer(randomPlayer) {
        this.#activePlayerView = new RandomPlayerView(randomPlayer);
    }
}

export { TurnView }

File: tech-java-orientadoObjetos/connect4Old/views/TurnView.java
package views;

import models.HumanPlayer;
import models.MinMaxPlayer;
import models.PlayerVisitor;
import models.RandomPlayer;
import models.Turn;
import utils.ClosedIntervalDialog;

public class TurnView implements PlayerVisitor {
    private Turn turn;
    private PlayerView activePlayerView;

    public TurnView(Turn turn) {
        super();
        this.turn = turn;
    }
    public void getNumberOfHumanPlayers() { 
        ClosedIntervalDialog closedIntervalDialog = new ClosedIntervalDialog(0, this.turn.getNumberPlayers());
        closedIntervalDialog.read(Message.NUM_PLAYERS.toString());
        this.turn.reset(closedIntervalDialog.getAnswer());
    }
    public void getMachineTypePlayer() { 
        ClosedIntervalDialog closedIntervalDialog = new ClosedIntervalDialog(1, 2);
        closedIntervalDialog.read(Message.TYPE_MACHINE.toString());
        this.turn.setTypeMachine(closedIntervalDialog.getAnswer());
    }

    public void play() {
        this.turn.getActivePlayer().accept(this);
        this.turn.play(this.activePlayerView.getColumn());
    }

    public void writeResult() {
        if ((this.turn.getBoard()).isWinner()) {
            this.activePlayerView.writeWinner();
        } else {
            Message.PLAYERS_TIED.writeln();
        }
    }
    public void visit(HumanPlayer humanPlayer) {
        this.activePlayerView = new HumanPlayerView(humanPlayer);
    }
    public void visit(RandomPlayer randomPlayer) {
        this.activePlayerView = new RandomPlayerView(randomPlayer);
    }
    public void visit(MinMaxPlayer minMaxPlayer) {
        this.activePlayerView = new MinMaxPlayerView(minMaxPlayer);
    }
}
